query_tests/src/scenarios/delete.rs

```diff
@@ -618,1096 +930,291 @@ impl DbSetup for ThreeDeleteThreeChunks {
-    let db = make_db().await.db;
-    // create an open MUB
-    write_lp(&db, &lp_lines.join("\n")).await;
-    // One open MUB, no RUB, no OS
-    assert_eq!(count_mutable_buffer_chunks(&db), 1);
-    assert_eq!(count_read_buffer_chunks(&db), 0);
-    assert_eq!(count_object_store_chunks(&db), 0);
-    db.delete("cpu", Arc::new(pred)).await.unwrap();
-    // Still one but frozen MUB, no RUB, no OS
-    assert_eq!(count_mutable_buffer_chunks(&db), 1);
-    assert_eq!(count_read_buffer_chunks(&db), 0);
-    assert_eq!(count_object_store_chunks(&db), 0);
-
-    DbScenario {
-        scenario_name: "Deleted data in MUB".into(),
-        db,
-    }
-}
+// Helper structs and functions
 
-async fn make_delete_mub_delete(
-    lp_lines: Vec<&str>,
-    pred1: Predicate,
-    pred2: Predicate,
-) -> DbScenario {
-    let db = make_db().await.db;
-    // create an open MUB
-    write_lp(&db, &lp_lines.join("\n")).await;
-    // One open MUB, no RUB, no OS
-    assert_eq!(count_mutable_buffer_chunks(&db), 1);
-    assert_eq!(count_read_buffer_chunks(&db), 0);
-    assert_eq!(count_object_store_chunks(&db), 0);
-    // delete from MUB
-    db.delete("cpu", Arc::new(pred1)).await.unwrap();
-    // Still one but frozen MUB, no RUB, no OS
-    assert_eq!(count_mutable_buffer_chunks(&db), 1);
-    assert_eq!(count_read_buffer_chunks(&db), 0);
-    assert_eq!(count_object_store_chunks(&db), 0);
-    // delete from frozen MUB
-    db.delete("cpu", Arc::new(pred2)).await.unwrap();
-    // Still one frozen MUB, no RUB, no OS
-    assert_eq!(count_mutable_buffer_chunks(&db), 1);
-    assert_eq!(count_read_buffer_chunks(&db), 0);
-    assert_eq!(count_object_store_chunks(&db), 0);
-
-    DbScenario {
-        scenario_name: "Deleted data from MUB then move and then delete data from frozen MUB"
-            .into(),
-        db,
-    }
+#[derive(Debug, Clone)]
+pub struct ChunkData<'a> {
+    lp_lines: Vec<&'a str>,
+    chunk_type: ChunkType,
 }
 
-async fn make_delete_mub_to_rub(
-    lp_lines: Vec<&str>,
-    pred: Predicate,
-    table_name: &str,
-    partition_key: &str,
-) -> DbScenario {
-    let db = make_db().await.db;
-    // create an open MUB
-    write_lp(&db, &lp_lines.join("\n")).await;
-    // delete data in MUB
-    db.delete("cpu", Arc::new(pred)).await.unwrap();
-    // move MUB to RUB and the delete predicate will be automatically included in RUB
-    db.rollover_partition(table_name, partition_key)
-        .await
-        .unwrap();
-    db.move_chunk_to_read_buffer(table_name, partition_key, ChunkId::new(0))
-        .await
-        .unwrap();
-    // No MUB, one RUB, no OS
-    assert_eq!(count_mutable_buffer_chunks(&db), 0);
-    assert_eq!(count_read_buffer_chunks(&db), 1);
-    assert_eq!(count_object_store_chunks(&db), 0);
-
-    DbScenario {
-        scenario_name: "Deleted data in RUB moved from MUB".into(),
-        db,
-    }
+#[derive(Debug, Clone)]
+pub enum ChunkType {
+    Mubo, // open MUB
```

Comment:
Would it be possible to make these doc comments (so when I hover over them in my editor the definition of Mubo appears. Something like

```suggestion
    /// Open MUB
    Mubo, 
```

Also, I felt Mubo sounded like a perfect name for a startup and turns out there is already one: https://www.mubo-app.com/ 😆